File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    for (int i = 1;; i++) {
        for (int j = i;; j++) {
            if ((i + 2) * (j + 2) > brown + yellow) break;
            if (i * j > yellow) break;
            else if (i * j == yellow && (i + 2) * (j + 2) == brown + yellow) {
                answer.push_back(j + 2);
                answer.push_back(i + 2);
            }
        }
        if (!answer.empty()) break;
    }
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.kt
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        var answer = IntArray(2)
        var loop = true
        
        if (isPerfectSquare(yellow)) {
            answer[0] = yellow + 2
            answer[1] = yellow + 2
        } else {            
            while(loop) {
                for (i in yellow downTo 2) {
                    if (yellow % i != 0) continue
                    
                    val width = i
                    val height = yellow / i
                    
                    if ((width + 2) * (height + 2) == brown + yellow) {
                        answer[0] = width + 2
                        answer[1] = height + 2
                        loop = false
                        break
                    }
                }
            }
        }
        
        return answer
    }
    
    fun isPerfectSquare(n: Int): Boolean {
        val sqrt = Math.sqrt(n.toDouble())
        return n.toDouble() == sqrt.toInt().toDouble()
    }
}